test/support/fixture_helper.rb

```diff
@@ -0,0 +1,133 @@
+# typed: ignore
+# frozen_string_literal: true
+
+require "rails_test_helper"
+
+module FixtureHelper
+  TEMP_FIXTURE_DIR = ROOT.join("tmp", "fixtures").to_s
+  FileUtils.mkdir_p(TEMP_FIXTURE_DIR)
+
+  Minitest.after_run do
+    # This does not get called on early exit, e.g. `Ctrl-C` or `(byebug) exit`.
+    FileUtils.remove_entry(TEMP_FIXTURE_DIR, true)
+  end
+
+  def around
+    old_working_dir = Dir.pwd
+    cache_rails_paths
+
+    yield
+
+    restore_rails_paths
+    Dir.chdir(old_working_dir)
+    FileUtils.remove_entry(@app_dir, true) if defined? @app_dir
+  end
+
+  def copy_template(template)
+    copy_dir("test/fixtures/#{template}")
+    Dir.chdir(app_dir)
+
+    case template
+    when :minimal
+      rails_autoload_paths(paths_to("/components/sales/app/models"))
```

Comment:
This sucky part is where we reconfigure the `Rails.application` singleton's bits that Packwerk cares about.

Reply:
This should be relevant only for the test for the application validator. No other test in packwerk should depend on anything in Rails.

Reply:
ApplicationValidator gets the paths from Configuration.  A test author could provide the correct temporary directory load paths through the configuration object as long as they know exactly what those paths are:
```ruby
def fixture_load_paths(template = :minimal)
  case template
  when :minimal
    paths_to("/components/sales/app/models")
  when :skeleton
    paths_to(
      "components/sales/app/models",
      "components/platform/app/models",
      #...
      "vendor/cache/gems/example/models"
    )
  else
    raise SomeError
  end
end
# ...
test "Something that cares about load paths" do
  config_params = read_fixture_packerk_yml_as_hash.merge(
    load_paths: fixture_load_paths(:the_fixure),
    probably_other_params: as_well
  )
  config = Packwerk::Configuration.new(config_params, config_path: app_dir)
  # ...
end
```

That decouples tests that are sensitive to load paths from the Rails singleton.  But yuck.

ApplicationValidator tests care about load paths, as you said.  The test suite for Configuration also cares, and it benefits from my proposed fixture helper setup/teardown.  To a much lesser degree, some of the Cli's tests are affected by load paths.  We currently set up that context via mocking.

I feel that the FixtureHelper's value is in setting up a complete, coherent runtime sandbox for a given test.  That includes a proper application directory, current working directory, and Rails singleton.  Although most tests will not be impacted by coherent Rails paths, test authors get to work in a setting where calling `Configuration.new` or `Configuration.from_path(app_dir)` will just work.